python/CMakeLists.txt

```diff
@@ -50,6 +50,12 @@ if (BUILD_PYTHON_PIP_PACKAGE)
   # configure_file(${CMAKE_CURRENT_SOURCE_DIR}/openstudio_dynamic.py ${PROJECT_BINARY_DIR}/Products/python/openstudio_dynamic.py COPYONLY)
   configure_file(${CMAKE_CURRENT_SOURCE_DIR}/openstudio.py ${PYTHON_PACKAGE_FOLDER}/openstudio/__init__.py COPYONLY)
 
+  if(PYTHON_PIP_REPOSITORY STREQUAL "testpypi")
+    execute_process( COMMAND ${Python_EXECUTABLE} "${CMAKE_CURRENT_SOURCE_DIR}/find_pypi_tag.py" OUTPUT_VARIABLE PYPI_VERSION)
+  else()
+    execute_process( COMMAND ${Python_EXECUTABLE} "${CMAKE_CURRENT_SOURCE_DIR}/ci/find_pypi_tag.py" "--pypi" OUTPUT_VARIABLE PYPI_VERSION)
+  endif()
+  message("On Repository '${PYTHON_PIP_REPOSITORY}', computed next available version as PYPI_VERSION=${PYPI_VERSION}")
```

Comment:
Call the python script to figure out the next available release you can use for upload.

if our CMakeLists.txt list 3.1.1-alpha for eg, it'll check that testpypi already has up to 3.1.1a14, and make it 3.1.1a15

Reply:
This versioning scheme should work fine for develop branches.  You commented on this, but users can call `openstudio; openstudio.openStudioLongVersion` to get the SHA if need be since PyPis doesn't offer a good way to include that in the version.